.github/workflows/build-publish-mcr.yml

```diff
@@ -6,63 +6,107 @@ permissions:
+        - os: windows-2022
+          file: ./builder/Dockerfile.windows
+          baseimage: 'mcr.microsoft.com/windows/nanoserver:ltsc2022'
+          tagext: 'nanoserver2022'
+    defaults:
+      run:
+        shell: pwsh
+    steps:
+    - uses: actions/checkout@v3
+    - name: 'Az CLI login'
+      uses: azure/login@v1
+      with:
+        client-id: ${{ vars.AZURE_CLIENT_ID }}
+        tenant-id: ${{ vars.AZURE_TENANT_ID }}
+        subscription-id: ${{ vars.AZURE_SUBSCRIPTION_ID }}
+    - name: 'Publish to ACR'
+      id: publish
+      run: |
+        $tag = "${{ needs.common.outputs.tagbase }}-${{ matrix.tagext }}"
+        echo "tag-${{ matrix.os }}=$tag" >> $env:GITHUB_OUTPUT
+        docker build -f ${{ matrix.file }} --build-arg BASE_IMAGE=${{ matrix.baseimage }} -t $tag .
+        az acr login -n ${{ vars.AZURE_REGISTRY_SERVER }}
+        docker push $tag
+    outputs:
+      linux: ${{ steps.publish.outputs.tag-ubuntu-latest }}
+      win2019: ${{ steps.publish.outputs.tag-windows-2019 }}
+      win2022: ${{ steps.publish.outputs.tag-windows-2022 }}
+  update-manifest:
+    runs-on: ubuntu-latest
+    needs: [common, publish]
     defaults:
       run:
         shell: pwsh
     steps:
-      - uses: actions/checkout@v2
-      - name: Get Changelog Entry
-        id: changelog_reader
-        uses: mindsers/changelog-reader-action@v2
-        with:
-          validation_depth: 10
-          path: ./CHANGELOG.md
-      - name: Display output
-        run: |
-           echo "Version: ${{ steps.changelog_reader.outputs.version }}"
-           echo "Changes: ${{ steps.changelog_reader.outputs.changes }}"
-      - name: 'Az CLI login'
-        uses: azure/login@v1
-        with:
-          client-id: ${{ secrets.AZURE_CLIENT_ID }}
-          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
-          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
-      - name: 'Run Azure CLI commands'
-        run: |
-          docker build -f ${{ matrix.file }} -t ${{ secrets.AZURE_REGISTRY_SERVER }}/public/${{ matrix.imagename }}:${{ steps.changelog_reader.outputs.version }} .
-          az acr login -n ${{ secrets.AZURE_REGISTRY_SERVER }}
-          docker push ${{ secrets.AZURE_REGISTRY_SERVER }}/public/${{ matrix.imagename }}:${{ steps.changelog_reader.outputs.version }}
-          echo "acr push done"
+    - uses: actions/checkout@v3
+    - name: 'Az CLI login'
+      uses: azure/login@v1
+      with:
+        client-id: ${{ vars.AZURE_CLIENT_ID }}
+        tenant-id: ${{ vars.AZURE_TENANT_ID }}
+        subscription-id: ${{ vars.AZURE_SUBSCRIPTION_ID }}
+    - name: Publish cross-platform manifest
+      id: publish
+      run: |
+        $tagbase = "${{ needs.common.outputs.tagbase }}"
+        az acr login -n ${{ vars.AZURE_REGISTRY_SERVER }}
+        docker manifest create $tagbase ${{ needs.publish.outputs.linux }} ${{ needs.publish.outputs.win2019 }} ${{ needs.publish.outputs.win2022 }}
+        docker manifest push $tagbase
```

Comment:
💡 for historic Archival reasons: this command is experimental for now: https://docs.docker.com/engine/reference/commandline/manifest_push/ 

<img width="1097" alt="Screenshot 2023-01-25 at 2 01 08 PM" src="https://user-images.githubusercontent.com/6233295/214515451-a09e6cc7-656b-458e-a712-b9c420cac769.png">

Reply:
Interesting! I didn't notice that. I used it because it was documented [here](https://learn.microsoft.com/en-us/azure/container-registry/push-multi-architecture-images#push-a-multi-arch-image) as the recommended way to publish multi-arch images. I'm prepared to take a punt on this functionality not being removed, and handle it if the command/syntax changes.